native/dlaqr5.go

```diff
@@ -0,0 +1,609 @@
+				}
+				if accum {
+					kms := k - incol
+					for j := max(1, ktop-incol) - 1; j < kdu; j++ {
+						refsum := v[m22*ldv] * (u[j*ldu+kms] + v[m22*ldv+1]*u[j*ldu+kms+1])
+						u[j*ldu+kms] -= refsum
+						u[j*ldu+kms+1] -= refsum * v[m22*ldv+1]
+					}
+				} else if wantz {
+					for j := iloz; j <= ihiz; j++ {
+						refsum := v[m22*ldv] * (z[j*ldz+k+1] + v[m22*ldv+1]*z[j*ldz+k+2])
+						z[j*ldz+k+1] -= refsum
+						z[j*ldz+k+2] -= refsum * v[m22*ldv+1]
+					}
+				}
+			}
+
+			// Vigilant deflation check.
+			mstart := mtop
+			if krcol+3*mstart < ktop {
+				mstart++
+			}
+			mend := mbot
+			if bmp22 {
+				mend++
+			}
+			if krcol == kbot-2 {
+				mend++
+			}
+			for m := mstart; m <= mend; m++ {
+				k := min(kbot-1, krcol+3*m)
+
+				// The following convergence test requires that the tradition
+				// small-compared-to-nearby-diagonals criterion and the Ahues &
+				// Tisseur (LAWN 122, 1997) criteria both be satisfied. The latter
+				// improves accuracy in some examples. Falling back on an alternate
+				// convergence criterion when tst1 or tst2 is zero (as done here) is
+				// traditional but probably unnecessary.
+
+				if h[(k+1)*ldh+k] == 0 {
+					continue
+				}
+				tst1 := math.Abs(h[k*ldh+k]) + math.Abs(h[(k+1)*ldh+k+1])
+				if tst1 == 0 {
+					if k >= ktop+1 {
+						tst1 += math.Abs(h[k*ldh+k-1])
+					}
+					if k >= ktop+2 {
+						tst1 += math.Abs(h[k*ldh+k-2])
+					}
+					if k >= ktop+3 {
+						tst1 += math.Abs(h[k*ldh+k-3])
+					}
+					if k <= kbot-2 {
+						tst1 += math.Abs(h[(k+2)*ldh+k+1])
+					}
+					if k <= kbot-3 {
+						tst1 += math.Abs(h[(k+3)*ldh+k+1])
+					}
+					if k <= kbot-4 {
+						tst1 += math.Abs(h[(k+4)*ldh+k+1])
+					}
+				}
+				if math.Abs(h[(k+1)*ldh+k]) <= math.Max(smlnum, ulp*tst1) {
+					h12 := math.Max(math.Abs(h[(k+1)*ldh+k]), math.Abs(h[k*ldh+k+1]))
+					h21 := math.Min(math.Abs(h[(k+1)*ldh+k]), math.Abs(h[k*ldh+k+1]))
+					h11 := math.Max(math.Abs(h[(k+1)*ldh+k+1]), math.Abs(h[k*ldh+k]-h[(k+1)*ldh+k+1]))
+					h22 := math.Min(math.Abs(h[(k+1)*ldh+k+1]), math.Abs(h[k*ldh+k]-h[(k+1)*ldh+k+1]))
+					scl := h11 + h12
+					tst2 := h22 * (h11 / scl)
+					if tst2 == 0 || h21*(h12/scl) <= math.Max(smlnum, ulp*tst2) {
+						h[(k+1)*ldh+k] = 0
+					}
+				}
+			}
+
+			// Fill in the last row of each bulge.
+			mend = min(nbmps, (kbot-krcol-1)/3) - 1
+			for m := mtop; m <= mend; m++ {
+				k := krcol + 3*m
+				refsum := v[m*ldv] * v[m*ldv+2] * h[(k+4)*ldh+k+3]
+				h[(k+4)*ldh+k+1] = -refsum
+				h[(k+4)*ldh+k+2] = -refsum * v[m*ldv+1]
+				h[(k+4)*ldh+k+3] -= refsum * v[m*ldv+2]
+			}
+		}
+
+		// Use U (if accumulated) to update far-from-diagonal entries in H.
+		// If required, use U to update Z as well.
+		if !accum {
+			continue
+		}
+		var jtop, jbot int
+		if wantt {
+			jtop = 0
+			jbot = n - 1
+		} else {
+			jtop = ktop
+			jbot = kbot
+		}
+		bi := blas64.Implementation()
+		if !blk22 || incol < ktop || ndcol > kbot || ns <= 2 {
```

Comment:
I don't think this will affect performance. I changed to `incol < ktop || kbot < ndcol`, the order from lower to higher indices seems more natural.